libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigationProvider.kt

```diff
@@ -18,6 +21,9 @@ object MapboxNavigationProvider {
      * @param navigationOptions
      */
     @JvmStatic
+    @Deprecated(
+        message = "Set the navigation options with MapboxNavigationApp.setup"
+    )
```

Comment:
Let's provide `ReplaceWith` annotation to allow IDEs to automatically apply the replacements. ([here an example](
https://github.com/mapbox/mapbox-navigation-android/blob/47e17155e6766a84b6de012d8e6874d65378ffcd/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSource.kt#L395))

Reply:
I was playing with `ReplaceWith` but it does not work. There is not a 1:1 function migration from `MapboxNavigationProvider` to `MapboxNavigationApp`